op-node/rollup/derive/engine_controller.go

```diff
@@ -176,7 +176,12 @@ func (e *EngineController) ConfirmPayload(ctx context.Context, agossip async.Asy
 	if e.buildingInfo == (eth.PayloadInfo{}) && agossip.Get() == nil {
 		return nil, BlockInsertPrestateErr, fmt.Errorf("cannot complete payload building: not currently building a payload")
 	}
-	if e.buildingOnto.Hash != e.unsafeHead.Hash { // E.g. when safe-attributes consolidation fails, it will drop the existing work.
+	if p := agossip.Get(); p != nil && e.buildingOnto == (eth.L2BlockRef{}) {
+		e.log.Warn("found reusable payload from async gossiper, and buildingOnto is empty. Reusing payload",
```

Comment:
```suggestion
		e.log.Warn("Found reusable payload from async gossiper, and no block was being built. Reusing payload.",
```

Nit: describe behaviors rather than internal details like `buildingOnto`, and use punctuation since it already is a sentence (and logs are not in Go error formatting).